app/src/main/java/com/example/ramen/clashroyaleupdates/WebViewActivity.java

```diff
@@ -1,54 +1,67 @@
 package com.example.ramen.clashroyaleupdates;
 
-import android.app.Activity;
-import android.content.Intent;
 import android.os.Bundle;
+import android.support.v7.app.ActionBar;
 import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
 import android.view.Menu;
-import android.view.MenuInflater;
 import android.view.MenuItem;
 import android.webkit.WebView;
-import android.support.v7.widget.Toolbar;
 
+import com.example.ramen.clashroyaleupdates.helper.Constants;
 import com.example.ramen.clashroyaleupdates.helper.Util;
 
 public class WebViewActivity extends AppCompatActivity {
 
-    WebView mWebPage;
+    private String TAG = getClass().getSimpleName();
+
+    private WebView mWebPage;
     private String link;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.web_page);
+
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setTitle("Clash Royale");
+        }
+
         mWebPage = (WebView) findViewById(R.id.wv_page);
-//        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
-//
-//        if (toolbar != null){
-//            setSupportActionBar(toolbar);
-//        }
 
         Bundle extras = getIntent().getExtras();
         if (extras == null) {
+            Log.e(TAG, "No data specified for webview");
+            return;
+        }
+        link = extras.getString(Constants.Intent.SELECTED_LINK);
+        if (link == null || link.trim().isEmpty()) {
+            Log.e(TAG, "No URL specified for webview");
             return;
         }
-        link = extras.getString("link");
         Util.openInWebView(mWebPage, link);
     }
 
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
-        new MenuInflater(this).inflate(R.menu.web, menu);
```

Comment:
This was the one causing the error. This creates a new menu inflater context rather than using the existing one. getMenuInflater() was correctly used in MainActivity, why use new here